method.rb

```diff
@@ -0,0 +1,248 @@
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i]
+      i += 1
+    end
+    output_array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i], i
+      i += 1
+    end
+    output_array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    output_array = []
+    my_each do |item|
+      output_array << item if (yield item) == true
+    end
+    output_array
+  end
+
+  def parameter_condition?(parameter)
+    parameter.is_a?(Regexp) || parameter.is_a?(Class) || parameter.nil?
+  end
+
+  def my_all?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
```

Comment:
**my_all** when no block or argument is given it should return true when none of the collection members are false or nil